### PR TITLE
When converting to raw strings, do not change \r\n sequences if not explicitly requested by the user

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/70209

### CODE REVIEW

File: src/Features/CSharp/Portable/ConvertToRawString/ConvertToRawStringHelpers.cs [hunk 1]
```diff
@@ -75,10 +75,28 @@ public static bool IsInDirective(SyntaxNode? node)
         return false;
     }
 
-    public static bool CanConvert(VirtualCharSequence characters)
-        => !characters.IsDefault && characters.All(static ch => CanConvert(ch));
+    public static bool CanConvert(VirtualCharSequence characters, out bool containsEscapedEndOfLineCharacter)
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/ConvertToRawString/ConvertToRawStringHelpers.cs:78]: >containsEscapedEndOfLineCharacter [](http://example.com/codeflow?start=75&length=33)

It's ok for containsEscapedEndOfLineCharacter to have the wrong value if this method returns false?

File: src/Features/CSharp/Portable/ConvertToRawString/ConvertInterpolatedStringToRawStringCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -178,7 +184,7 @@ protected override InterpolatedStringExpressionSyntax Convert(
         SyntaxFormattingOptions formattingOptions,
         CancellationToken cancellationToken)
     {
-        if (kind == ConvertToRawKind.SingleLine)
+        if (kind.HasFlag(ConvertToRawKind.SingleLine))
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/ConvertToRawString/ConvertInterpolatedStringToRawStringCodeRefactoringProvider.cs:187]: >HasFlag [](http://example.com/codeflow?start=17&length=7)

Is this executed on the VS side? If so, will this method be called a lot. HasFlag is notoriously slow on net fx.

File: src/Features/CSharp/Portable/ConvertToRawString/ConvertStringToRawStringCodeRefactoringProvider.cs [hunk 3]
```diff
@@ -22,24 +23,22 @@
...
-
-    [ImportingConstructor]
-    [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-    public ConvertStringToRawStringCodeRefactoringProvider()
+    static ConvertStringToRawStringCodeRefactoringProvider()
     {
+        var count = (int)ConvertToRawKind.ContainsEscapedEndOfLineCharacter * 2;
+        s_kindToEquivalenceKeyMap = new string[count];
+        for (var i = 0; i < count; i++)
+            s_kindToEquivalenceKeyMap[i] = i.ToString(CultureInfo.InvariantCulture);
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/ConvertToRawString/ConvertStringToRawStringCodeRefactoringProvider.cs:37]: >i.ToString [](http://example.com/codeflow?start=43&length=10)

Need more coffee, isn't this just putting string numbers in the array, and not the nameof the corresponding conversion?

File: src/Features/CSharp/Portable/ConvertToRawString/ConvertInterpolatedStringToRawStringCodeRefactoringProvider.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/ConvertToRawString/ConvertInterpolatedStringToRawStringCodeRefactoringProvider.cs:187]: It won't be called a lot. It's called only when you're on a raw string, and only a tiny number of times then.

File: src/Features/CSharp/Portable/ConvertToRawString/ConvertToRawStringHelpers.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/ConvertToRawString/ConvertToRawStringHelpers.cs:78]: Yes. If we can't convert, we bail out. So all that matters is the value when this returns true.

File: src/Features/CSharpTest/ConvertToRawString/ConvertRegularStringToRawString_FixAllTests.cs [hunk 4]
```diff
@@ -971,4 +966,75 @@ class Y
...
+            """
+            class C
+            {
+                void M()
+                {
+                    var description = @"
+
+            ";
+
+                    var second = Regex.Replace(description, {|FixAllInDocument:|}"(\r\n)", "$1$1");
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/CSharpTest/ConvertToRawString/ConvertRegularStringToRawString_FixAllTests.cs:1017]: Not sure I understand this comment

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**